### PR TITLE
Add build base, refactor base image build

### PR DESCRIPTION
**Pull request by csrwng**
Creates a separate base image for builds. The build-base-images pipeline now builds 2 images:
- test-base
- build-base

The source build will now use `test-base` as its base image.

### CODE REVIEW

**Comment by csrwng** [comment]
@stevekuznetsov

File: cluster/ci/origin/images/build-base/run.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+if [[ -n "${UMASK}" ]]; then
+	umask "${UMASK}" && "$@"
```
Comment (stevekuznetsov) [inline cluster/ci/origin/images/build-base/run.sh]: Sorry, I forget -- why did we need this indirection?

File: cluster/ci/origin/pipelines/base-images/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "RELEASE_SRC_URL=${params.RELEASE_SRC_URL}",
+            "RELEASE_SRC_REF=${params.RELEASE_SRC_REF}",
+            "ORIGIN_SRC_URL=${params.ORIGIN_URL}",
+            "IMAGE_SECRET=${imageSecret}")
+          def testBaseBuild = createTemplate(this, "${templateBase}/test-base-build.yaml",
+            "RELEASE_SRC_URL=${params.RELEASE_SRC_URL}",
+            "RELEASE_SRC_REF=${params.RELEASE_SRC_REF}",
+            "ORIGIN_SRC_URL=${params.ORIGIN_URL}",
+            "IMAGE_SECRET=${imageSecret}")
+          // For now wait for builds serially. Parallel blocks
```
Comment (stevekuznetsov) [inline cluster/ci/origin/pipelines/base-images/Jenkinsfile]: :( :(

Comment (csrwng) [reply cluster/ci/origin/pipelines/base-images/Jenkinsfile]: The builds run in parallel, we just wait for them serially. I don't think it's that bad. But basically groovy does some crazy things when running parallel blocks... variables from one block pollute the other block, etc.

File: cluster/ci/origin/images/build-base/run.sh (hunk 1 above)
Comment (csrwng) [reply cluster/ci/origin/images/build-base/run.sh]: Just to not have umask explicitly called wherever we run something... more of a hygiene thing.

**Pull request merged**